src/datetime/tests.rs

```diff
@@ -1486,3 +1486,36 @@ fn locale_decimal_point() {
     assert_eq!(dt.format_localized("%T%.6f", ar_SY).to_string(), "18:58:00.123456");
     assert_eq!(dt.format_localized("%T%.9f", ar_SY).to_string(), "18:58:00.123456780");
 }
+
+/// This is an extended test for <https://github.com/chronotope/chrono/issues/1289>.
+#[test]
+fn nano_roundrip() {
+    const BILLION: i64 = 1_000_000_000;
+
+    for nanos in [
+        i64::MIN,
+        i64::MIN + 1,
+        i64::MIN + 2,
+        i64::MIN + BILLION - 1,
+        i64::MIN + BILLION,
+        i64::MIN + BILLION + 1,
+        -BILLION - 1,
+        -BILLION,
+        -BILLION + 1,
+        0,
+        BILLION - 1,
+        BILLION,
+        BILLION + 1,
+        i64::MAX - BILLION - 1,
+        i64::MAX - BILLION,
+        i64::MAX - BILLION + 1,
+        i64::MAX - 2,
+        i64::MAX - 1,
+        i64::MAX,
+    ] {
+        println!("nanos: {nanos}");
```

Comment:
Maybe remove this `println!`? Or can you change it so that the test compiles on Rust 1.57 (our MSRV)?

Reply:
Just needs `println!("nanos: {}", nanos)`.

Reply:
Adjusted the println to be compatible w/ Rust 1.57. I think keeping it helps debugging a failing tests due to the many cases[^1].

[^1]: I wish there would be a slightly nicer story for parametrized tests in "standard Rust" though.

Reply:
(Hopefully [there will be](https://epage.github.io/blog/2023/06/iterating-on-test/) in the not-too-distant future.)